src/s2python/generated/gen_s2.py

```diff
@@ -214,7 +214,7 @@ class Config:
         ..., description="ID of this instruction (as provided by the CEM) "
     )
     message_id: ID = Field(..., description="ID of this message")
-    message_type: str = Field("InstructionStatusUpdate", const=True)
+    message_type: Literal["InstructionStatusUpdate"] = Field(default="InstructionStatusUpdate")
```

Comment:
No issues with moving from a pydantic const to the `Literal` type. However, datamodel-codegen does not support to use it for pydantic v1 output. Prioritising updating to pydantic v2 so it is automatically generated. https://github.com/flexiblepower/s2-python/issues/10

Reply:
PR is ready that has taken care of generating Literals instead of constants by moving to pydantic v2: #40

Reply:
HomeAssistant is not yet compatible with Pydantic 2. We are tracking https://github.com/home-assistant/core/issues/99218 to see when it is, but until then I'm not sure of the implications with regards to compatibility if the `s2-python` package would already migrate to Pydantic 2. Is the use of a `Literal` problematic at this point, or do you only regard it as a less preferred option over moving to Pydantic first.

Reply:
HI @Flix6x ! gen_s2.py is the output of the datamodel-codegen code generator. As such, we can move to literals but in its current shape it would be modifying a generated code file manually or as a post-processing step.

Perhaps we should split the discussion. We could discuss 'Why and when to move to pydantic v2' here: https://github.com/flexiblepower/s2-python/issues/10

Then we can keep the discussion in this comment thread to how to close this PR. Moving to pydantic v2 solves this automatically as datamodel-codegen uses the `Literal` type in its output for `pydantic-v2` output. So I was expecting that moving to pydantic v2 would solve this issue without any extra work. However, if we want to move to `Literal` without moving to pydantic v2, I would like to propose an alternative. Manually altering generated files is error prone. We could move the `Literal` lines from the generated code that are introduced by this PR to the non-generated classes, essentially overwriting the fields as we have done with ID's. The referenced line of this comment refers to `InstructionStatusUpdate`. An example on where to move it (see the last line):

File: https://github.com/flexiblepower/s2-python/blob/main/src/s2python/common/instruction_status_update.py
```python

@catch_and_convert_exceptions
class InstructionStatusUpdate(
    GenInstructionStatusUpdate, S2Message["InstructionStatusUpdate"]
):
    class Config(GenInstructionStatusUpdate.Config):
        validate_assignment = True

    message_id: uuid.UUID = GenInstructionStatusUpdate.__fields__["message_id"].field_info  # type: ignore[assignment]
    instruction_id: uuid.UUID = GenInstructionStatusUpdate.__fields__[
        "instruction_id"
    ].field_info  # type: ignore[assignment]
    message_type: Literal["InstructionStatusUpdate"] = Field(default="InstructionStatusUpdate")  # <-- Line which is added
```

What do you think of this idea? In case you like it, perhaps its a good idea to first test with a small case to see if it has the intended effect.

Reply:
Originally what I had in mind was to manually overwrite the types that datamodel-codegen generate with manually added lines such as `message_type: Literal["InstructionStatusUpdate"] = Field(default="InstructionStatusUpdate")  # <-- Line which is added`. However, as you are taking over pydantic v1 branch, feel free to use another method which fits for your efforts.

Reply:
After consultation with @victorgarcia98:
- I understand better now that no changes on the generated classes were actually expected, because we are now overwriting the message type in their **sub**-classes.
- I realized I had been too enthousiastic with overwriting message types in this PR, because I had unknowingly also added a message type to classes that were not themselves S2 messages, but rather described elements contained in such messages.

Reply:
To move this PR forward, I propose I remove all changes relating to the `Literal` issue and let it only contain the addition of the `S2Message` class and the renaming of the old `S2Message` class into `S2MessageComponent`.

In https://github.com/flexiblepower/s2-python/pull/56, I propose to take ownership of releases from the `pydantic-v1` branch, using dev versioning.